### PR TITLE
feat: form items should be able to be disabled

### PR DESCRIPTION
**Pull request by fmamic**
Added optional disabled flag to the form items

# Description

The 'disabled' flag is added which allows form-items to be disabled.

## Motivation & context

JSON schema can contain 'disabled' field and a user can disable form-items.

Closes #1251 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [X] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [X] I have added tests for my changes.
- [X] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [X] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-form-generator-react/src/form/form-item.align-horizontal.spec.tsx [hunk 1]
```diff
@@ -60,6 +60,17 @@ describe("AlignHorizontal", () => {
         expect(handleChange).toHaveBeenCalled();
         expect(handleChange.mock.calls[0][1]).toEqual("left");
     });
+    test("should be disabled if disabled props is passed", () => {
+        const rendered: any = mount(
+            <AlignHorizontal {...alignHorizontalProps} disabled={true} />
+        );
+        const inputs: any = rendered.find("input");
```
Comment (chrisdholt) [inline packages/fast-form-generator-react/src/form/form-item.align-horizontal.spec.tsx]: I think we have a pattern of rendered being "any", but here it seems like you are finding HTMLInputElements. Is there a typing we can leverage here?

File: packages/fast-form-generator-react/src/form/form-item.align-vertical.spec.tsx [hunk 2]
```diff
@@ -60,6 +60,17 @@ describe("AlignVertical", () => {
         expect(handleChange).toHaveBeenCalled();
         expect(handleChange.mock.calls[0][1]).toEqual("top");
     });
+    test("should be disabled if disabled props is passed", () => {
+        const rendered: any = mount(
+            <AlignVertical {...alignVerticalProps} disabled={true} />
+        );
+        const inputs: any = rendered.find("input");
```
Comment (chrisdholt) [inline packages/fast-form-generator-react/src/form/form-item.align-vertical.spec.tsx]: Is there a typing we can use here? `HTMLInputElement[]`?

File: packages/fast-form-generator-react/src/form/form-item.align-horizontal.spec.tsx (hunk 1 above)
Comment (fmamic) [reply packages/fast-form-generator-react/src/form/form-item.align-horizontal.spec.tsx]: ShallowWrapper is returned from find(). I will add to dev dependencies @types/enzyme which can help with the typing.

**Comment by awentzel** [comment]
@fmamic looks like we're all good for you to Squash and merge in your changes.  Thanks for your contribution.

**Pull request merged**